contracts/AgentMech.sol

```diff
@@ -96,19 +115,31 @@ contract AgentMech is ERC721Mech {
 
         // Increase the number of undelivered requests
         numUndeliveredRequests++;
+        // Increase the total number of requests
+        numTotalRequests++;
 
         emit Request(msg.sender, requestId, data);
     }
 
+    /// @dev Performs actions before the delivery of a request.
+    /// @param data Self-descriptive opaque data-blob.
+    /// @return requestData Data for the request processing.
+    function _preDeliver(uint256, bytes memory data) internal virtual returns (bytes memory requestData) {
+        requestData = data;
+    }
```

Comment:
The agent can wrap any data along with the request data that can be parsed during the pre-deliver function. This gives more flexibility in actions needed before the request is delivered.